### PR TITLE
BUG: Allow empty chunksize in stata reader when using iterator

### LINKED ISSUE #37280: BUG: regression: pandas.read_stata(filename, iterator=True) raises ValueError
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
>>> import pandas
>>> # for example https://gitlab.com/ViDA-NYU/datamart/datamart/-/blob/master/tests/data/stata118.dta
>>> iterator = pandas.read_stata(stata_file_name, iterator=True)
>>> list(iterator)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "site-packages/pandas/io/stata.py", line 1523, in __next__
    raise ValueError(
ValueError: chunksize must be set to a positive integer to use as an iterator.
```

#### Problem description

`read_stata(filename, iterator=True)` no longer works in pandas 1.1.3. **It worked in pandas 1.0.5.**

#### Expected Output

DataFrame is loaded correctly

#### Output of ``pd.show_versions()``

<details>

```
INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.8.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-51-generic
Version          : #56-Ubuntu SMP Mon Oct 5 14:28:49 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.3
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : 3.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.14.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : 0.8.4
fastparquet      : None
gcsfs            : 0.7.1
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.5.3
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None
```

</details>

### PR DESCRIPTION
**Pull request by bashtage**
- [X] closes #37280 
- [X] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/stata.py [hunk 1]
```diff
@@ -1518,11 +1518,8 @@ def _read_strls(self) -> None:
             self.GSO[str(v_o)] = decoded_va
 
     def __next__(self) -> DataFrame:
-        if self._chunksize is None:
-            raise ValueError(
-                "chunksize must be set to a positive integer to use as an iterator."
-            )
-        return self.read(nrows=self._chunksize or 1)
+        self._chunksize = 1 if self._chunksize is None else self._chunksize
```
Comment (jreback) [inline pandas/io/stata.py:1521]: is there validation when chunsize is passed in?

**Comment by jreback** [comment]
can you add a what's new note for the precision loss change (1.1.4)

**Comment by jreback** [comment]
also pls merge master

**Comment by bashtage** [comment]
I rebased on master and added a what's new.

File: pandas/io/stata.py (hunk 1 above)
Comment (bashtage) [reply pandas/io/stata.py]: It is not checked when set. I've refactored the code to not rely on chunksize being set to determine if an iterator is being used.

Comment (jreback) [reply pandas/io/stata.py]: what i mean is what if someone passes `chunksize<=0` ? do we appropriately raise

**Comment by jreback** [comment]
small conflict and pls rebase

File: pandas/io/stata.py (hunk 1 above)
Comment (bashtage) [reply pandas/io/stata.py]: Yes, there is explicit  validation to ensure that it is an `int` that is `>0`

https://github.com/pandas-dev/pandas/blob/b2ea05f46fa9d65d8176ea0217276b7c56549924/pandas/io/stata.py#L1049-L1052

**Comment by jreback** [comment]
thanks @bashtage

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**